### DISCUSSION TITLE
How to use Password connector to generate token using password grant?

### DISCUSSION THREAD

**Discussion by surnaja239068** [dexidp/dex#2297] [Q&A]
On executing token URL using Curl I'm getting invalid credentails.

curl -i -k -X POST -d "client_id=ldap&client_secret=ZXhhbXBsZS1hcHAtc2VjcmV0&response_type=token&scope=openid&state=&grant_type=password&username=my-user&password=testing" http://127.0.0.1:5556/dex/token
HTTP/1.1 401 Unauthorized
Content-Length: 76
Content-Type: application/json
Date: Sun, 17 Oct 2021 08:21:15 GMT

{"error":"invalid_client","error_description":"Invalid client credentials."}

Here is  my connector configuraiton.
To the example [ldap config](https://github.com/dexidp/dex/blob/master/examples/ldap/config-ldap.yaml), added passwordConnector 

> oauth2:
>   skipApprovalScreen: true
>   passwordConnector: ldap

**Comment by jghiloni** [comment]
Doubt you still need this due to the age of the question, but if someone ELSE comes upon it, the answer is to include the client credentials as basic auth credentials, so your curl should look something like this (using some other curl flags to clean this up):

```
curl -i -k -X POST -u ldap:ZXhhbXBsZS1hcHAtc2VjcmV0 \
  -F 'response_type=token' -F 'scope=openid' -F 'state=' \
  -F 'grant_type=password' -F 'username=my-user' -F 'password=testing' \
  -H "Content-Type: application/x-www-form-urlencoded" \
  http://127.0.0.1:5556/dex/token
```
